examples/pubsub/server_pubsub_publisher_iop.c

```diff
@@ -75,7 +75,8 @@ addPubSubConnection(UA_Server *server, UA_String *transportProfile,
     connectionConfig.enabled = UA_TRUE;
     UA_Variant_setScalar(&connectionConfig.address, networkAddressUrl,
         &UA_TYPES[UA_TYPES_NETWORKADDRESSURLDATATYPE]);
-    connectionConfig.publisherId.numeric = Publisher_ID;
+    UA_UInt16 publisherId = Publisher_ID;
+    UA_Variant_setScalar(&connectionConfig.publisherId, &publisherId, &UA_TYPES[UA_TYPES_UINT16]);
```

Comment:
The old numeric implementation had a size of UInt32 but [encoded it as UInt16](https://github.com/open62541/open62541/blob/5dea0e3898ce87fa4c09d7a0e8dee6c2b35906f5/src/pubsub/ua_pubsub_networkmessage.c#L96) since it [set the Type incorrectly](https://github.com/open62541/open62541/blob/5dea0e3898ce87fa4c09d7a0e8dee6c2b35906f5/src/pubsub/ua_pubsub_writer.c#L1906). The subscribers in the examples expect UInt16 therefore. In the cases where the PublisherId didn't matter a random UInt32 was used so i used that aswell since the Type doesn't matter either.